docs/get-started/getting-started-on-prem.md

```diff
@@ -422,33 +422,218 @@ in these scenarios:
   - _GPU-to-GPU_: Shuffle blocks that were able to fit in GPU memory.
   - _Host-to-GPU_ and _Disk-to-GPU_: Shuffle blocks that spilled to host (or disk) but will be manifested 
   in the GPU in the downstream Spark task.
+  
+### System Setup
+
+In order to enable _RapidsShuffleManager_, UCX user-space libraries and its dependencies must be 
+installed on the host and within Docker containers. A host has additional requirements, like the 
+MLNX_OFED driver and `nv_peer_mem` kernel module.
+
+#### Baremetal
+
+1. If you have Mellanox hardware, please ensure you have the [MLNX_OFED driver](https://www.mellanox.com/products/infiniband-drivers/linux/mlnx_ofed), and the
+[`nv_peer_mem` kernel module](https://www.mellanox.com/products/GPUDirect-RDMA) installed. UCX packages
+   are compatible with MLNX_OFED 5.0+. Please install the latest driver available.
+
+   With `nv_peer_mem` (GPUDirectRDMA), IB/RoCE-based transfers can perform zero-copy transfers 
+   directly from GPU memory. Note that GPUDirectRDMA is known to show 
+   [performance and bugs](https://docs.nvidia.com/cuda/gpudirect-rdma/index.html#supported-systems) 
+   in machines that don't connect their GPUs and NICs to PCIe switches (i.e. directly to the 
+   root-complex). 
+   
+   If you encounter issues or poor performance, GPUDirectRDMA can be controlled via the 
+   UCX environment variable `UCX_IB_GPU_DIRECT_RDMA=no`, but please file a GitHub issue so we
```

Comment:
I suggest making `file a Github issue` a hyperlink to https://github.com/NVIDIA/spark-rapids/issues